packages/gated-content/src/conditions/index.ts

```diff
@@ -0,0 +1,137 @@
+  [type, value]: Entry<LeafCriterionFragment>,
+  env: EnvironmentConfig,
+): LitNestedAccessControlCondition<LitAccessControlCondition> {
+  switch (type) {
+    case AccessConditionType.Eoa:
+      return transformEoaCondition(value);
+    case AccessConditionType.Token:
+      return transformErc20Condition(value);
+    case AccessConditionType.Nft:
+      return transformNftCondition(value);
+    case AccessConditionType.Profile:
+      return transformProfileCondition(value, env);
+    case AccessConditionType.Collect:
+      return transformCollectCondition(value, env);
+    case AccessConditionType.Follow:
+      return transformFollowCondition(value, env);
+    default:
+      throw new InvariantError(`Unknown access criteria type: ${String(type)}`);
+  }
+}
+
+function transformCompoundCondition(
+  entry: Entry<AnyCriterionFragment>,
+  env: EnvironmentConfig,
+): LitNestedAccessControlCondition<LitAccessControlCondition> {
+  const [type, value] = entry;
+
+  if (type === AccessConditionType.And || type === AccessConditionType.Or) {
+    assertAtLeastTwoCriteria(value.criteria);
+    assertNoMoreThanFiveCriteria(value.criteria);
+
+    try {
+      const flat = flatten(
+        value.criteria.map((criterion) =>
+          transformSimpleCondition(extractConditionEntry(criterion), env),
+        ),
+      );
+      return insertObjectInBetweenArrayElements(flat, { operator: type });
+    } catch (err: unknown) {
+      if (err instanceof InvariantError) {
+        throw new InvalidAccessCriteriaError('Cannot nest conditions more than 2 levels deep.');
+      }
+      throw err;
+    }
+  }
+  return transformSimpleCondition(entry, env);
+}
+
+export function transform(
+  condition: AccessCondition,
+  env: EnvironmentConfig,
+): UnifiedAccessControlConditions {
+  const [type, value] = extractConditionEntry(condition);
+
+  if (type !== AccessConditionType.Or) {
+    throw new InvalidAccessCriteriaError('Root condition must be an OR condition');
+  }
+
+  if (value.criteria.length < 1) {
+    throw new InvalidAccessCriteriaError('Root condition must have at least one criteria');
+  }
+
+  assertNoMoreThanFiveCriteria(value.criteria);
+
+  if (value.criteria.length > 2) {
+    throw new InvalidAccessCriteriaError('Root conditions can only have up to 2 criteria.');
+  }
+
+  const rootEntries = value.criteria.map(extractConditionEntry);
+
+  if (!rootEntries.some(([t]) => t === AccessConditionType.Profile)) {
+    throw new InvalidAccessCriteriaError('Root conditions must contain a profile condition');
+  }
+
+  const flat = flatten(
+    rootEntries.map((entry) =>
+      transformCompoundCondition(entry as Entry<AnyCriterionFragment>, env),
+    ),
+  );
+
+  // the type assertion is needed because the Lit SDK typedef suggests nested conditions are not allowed but they are
+  return insertObjectInBetweenArrayElements(flat, {
+    operator: AccessConditionType.Or,
+  }) as UnifiedAccessControlConditions;
+}
```

Comment:
100% of the logic in this folder has been adapted from the `@lens-protocol/sdk-gated`. The validation logic has been merged into the traversing done during the `transform` logic, this reduced the times we need to 

Collect Access Condition when `thisPublication` is set assumes the `publicationId` has been provided by the caller. The Publication ID prediction logic needs to be done by the caller (`@lens-protocol/react` and `@lens-protocol/client`) cause they have significant differences on the way the next publication ID can be inferred.
Namely `@lens-protocol/react` requires to take into account what is in the current `TransactionQueue` while the `@lens-protocol/client` can take the traditional approach used in the `@lens-protocol/sdk-gated` package (just fetch last publication Id for the given profile and "add 1" to it).